### PR TITLE
Add Regtest mode for Litecoin

### PR DESCRIPTION
**Pull request by mrosseel**
Adding the regtest params for Litecoin, useful for dev testing without needing testnet coins.

### CODE REVIEW

File: core/src/test/java/org/libdohj/params/LitecoinRegTestParamsTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.bitcoinj.core.Block;
+import org.bitcoinj.core.Context;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ *
+ * @author Ross Nicoll
```
Comment (rnicoll) [inline core/src/test/java/org/libdohj/params/LitecoinRegTestParamsTest.java]: Can you change the author to yourself please? :)

File: core/src/main/java/org/libdohj/params/LitecoinRegTestParams.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    @Override
+    public boolean allowEmptyPeerChain() {
+        return true;
+    }
+
+    private static Block genesis;
+
+
+    /* Extract from Litecoin source code, definition of regtest params.
```
Comment (rnicoll) [inline core/src/main/java/org/libdohj/params/LitecoinRegTestParams.java]: I'm not sure it's a good idea to paste this in here; a link to the commit in Github (i.e. https://github.com/litecoin-project/litecoin/blob/edc66b374ea68107c721062152dd95e6aa037d53/src/chainparams.cpp ) would probably make more sense, as once we have this working the parameters shouldn't change unless the original changes (in which case the copied data is unhelpful).

**Review by rnicoll — CHANGES_REQUESTED** [review]
Couple of comments, mostly I'd rather not copy chunks of Litecoin code into libdohj as I don't think they're helpful once the code is developed. Other than that, looks good to go, thanks!

File: core/src/test/java/org/libdohj/params/LitecoinRegTestParamsTest.java (hunk 1 above)
Comment (mrosseel) [reply core/src/test/java/org/libdohj/params/LitecoinRegTestParamsTest.java]: ok

File: core/src/main/java/org/libdohj/params/LitecoinRegTestParams.java (hunk 2 above)
Comment (mrosseel) [reply core/src/main/java/org/libdohj/params/LitecoinRegTestParams.java]: yeah np, link is better

**Comment by mrosseel** [comment]
please recheck, thx

**Review by rnicoll — APPROVED** [review]

**Comment by rnicoll** [comment]
Many thanks for that!

**Pull request merged**